webui/react/src/utils/user.ts

```diff
@@ -29,14 +29,14 @@ export function isUserWithRoleInfo(
   return 'userId' in userOrGroup;
 }
 
-export function getName(userOrGroup: Readonly<UserOrGroup>): string {
+export function getName(userOrGroup: UserOrGroup): string {
   if (isUser(userOrGroup)) {
     return getDisplayName(userOrGroup);
   }
   return userOrGroup?.name ?? '';
 }
 
-export const getIdFromUserOrGroup = (userOrGroup: Readonly<UserOrGroup>): number => {
```

Comment:
Type narrowing doesnt work when type is wrapped by Readonly with type predicates since v5. It works fine with v4.9.5
[ts playground](https://www.typescriptlang.org/play?#code/PTAECcFMDMC5QBYBckAcDOsQHdcDokBPVSdAY3AEtUkAbAQwDsBzPAe3GeABM2z1gCJtwBGbNgGtgAJmCN64cG2yUWeZAFtaAKG1ESoAGKV0CUAF5QAb1DoVG+AAoAlBYB8oAG5tK3UAF8Abj1iSFAAIUpwP0sbaFpCJ1dzD29fAOD9MPC2JDNLY1NQAB8IqO5daABXRjIkSjZGUBNChEcSJHhWkrLo53gO5vQjEzMrbVAISCQq8Cb26dB6YdbnPDtKDVAAQnNLGu4YVUhuQNAQUEZc0GwOCRvKPNAAAwAiDY1X5qaO59BIR4ISDgUDafy6C5VdCqZhDUDVWqghF1BpNDRsTyQRxMTb0WjwHJ5VzjSaUaCgRwtUbYxi42jOYkTSZTGZzJa0jR49b2FzBSbgyZQVlNHGc2h4eKEXlgyo1FGNUDozHSGl0+AAJUg9F4jASAB5CQg3IzSeTKehWqqxQzrEzBdNZiKOVyPtL+UyhY72XSJQleecwHqALSgADqAAkAJoAfhl2kh0JY31AbBI4HoSA4SLl9QVSsgAH1VFa8QTcggTc0ze97F9VN7rbbmSyvaKXTznHyAh6HWy2+LJdLwdpkbm0RjC6oVf2NVqdfrDcam1WKTXNnWnXTK-bhQ325s3d2d63nQO-Z2wUA)

Reply:
You can solve this by overloading the guard function:
```typescript
function isFish(pet: Readonly<Fish | Bird>): pet is Readonly<Fish>
function isFish(pet: Fish | Bird): pet is Fish {
  return (pet as Fish).swim !== undefined; // not work with `"swim" in pet` either 
}
```